### PR TITLE
[Flink] Fix Multibyte Character Encoding in MultiTableCommittable Serialization

### LINKED ISSUE #3027: [Bug] Incorrect Handling of String Encoding in MultiTableCommittable Serialization
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

0.7

### Compute Engine

1.17

#### Issue description
When working with a `MultiTableCommittableSerializer` to handle serialized data, specifically metadata with non-ASCII characters (e.g., Chinese characters for table names), incorrect handling of byte buffers leads to missed characters or potential `BufferOverflowException` errors.

#### Steps to reproduce
The issue arises when deserializing byte arrays that represent strings with multibyte characters. For instance, deserializing metadata for tables with Chinese names can produce undesired errors or incorrect results.

### What doesn't meet your expectations?

The deserialization process should correctly convert byte arrays back to strings, accommodating multibyte characters without errors or data loss.

### Anything else?

Here is the stacktrace from the exception:
```
Caused by: java.nio.BufferOverflowException
	at java.nio.HeapByteBuffer.put(HeapByteBuffer.java:189)
	at java.nio.ByteBuffer.put(ByteBuffer.java:859)
	at org.apache.paimon.flink.sink.MultiTableCommittableSerializer.serialize(MultiTableCommittableSerializer.java:70)
	at org.apache.paimon.flink.sink.MultiTableCommittableSerializer.serialize(MultiTableCommittableSerializer.java:36)
	at org.apache.flink.core.io.SimpleVersionedSerialization.writeVersionAndSerialize(SimpleVersionedSerialization.java:59)
	at org.apache.flink.core.io.SimpleVersionedSerializerTypeSerializerProxy.serialize(SimpleVersionedSerializerTypeSerializerProxy.java:97)
	at org.apache.flink.streaming.runtime.streamrecord.StreamElementSerializer.serialize(StreamElementSerializer.java:165)
	at org.apache.flink.streaming.runtime.streamrecord.StreamElementSerializer.serialize(StreamElementSerializer.java:43)
```

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by Pandas886**
Linked issue: close #3027

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/MultiTableCommittableSerializer.java [hunk 1]
```diff
@@ -83,17 +84,18 @@ public MultiTableCommittable deserialize(int committableVersion, byte[] bytes)
         int databaseLen = buffer.getInt();
         byte[] databaseBytes = new byte[databaseLen];
         buffer.get(databaseBytes, 0, databaseLen);
-        String database = new String(databaseBytes);
+        String database = new String(databaseBytes,StandardCharsets.UTF_8);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/MultiTableCommittableSerializer.java:87]: `database.getBytes()` should use `database.getBytes(UTF_8)` too.

**Review by JingsongLi — COMMENTED** [review]
Hi @Pandas886, Thanks for your contribution!

Can we just use `DataOutputSerializer` here? To simplify serialization work.

**Comment by Pandas886** [comment]
> DataInputDeserializer

I'll try.

**Comment by Pandas886** [comment]
> Hi @Pandas886, Thanks for your contribution!您好，感谢您的贡献！
> 
> Can we just use `DataOutputSerializer` (with `DataInputDeserializer`) here? To simplify serialization work. `DataOutputSerializer` 1 `DataOutputSerializer` 1#简化序列化工作。

I'll try.

**Comment by JingsongLi** [comment]
I merged your work for 0.8.0 releasing.

**Pull request closed without merging**